src/LuaTransformer.ts

```diff
@@ -1369,15 +1371,20 @@ export class LuaTransformer {
         // Add object binding patterns
         let identifierIndex = 0;
         const bindingPatternDeclarations: tstl.Statement[] = [];
-        parameters.forEach(binding => {
+        parameters.forEach((binding, i) => {
             if (ts.isObjectBindingPattern(binding.name) || ts.isArrayBindingPattern(binding.name)) {
                 const identifier = tstl.createIdentifier(`____TS_bindingPattern${identifierIndex++}`);
                 bindingPatternDeclarations.push(...this.statementVisitResultToArray(
                     this.transformBindingPattern(binding.name, identifier)
                 ));
+                if (binding.initializer !== undefined) {
+                    const declaration = this.transformParameterDefaultValueDeclaration(identifier, binding.initializer);
+                    defaultValueDeclarations.splice(i, 0, declaration);
```

Comment:
I think this is is unnecessarily complex, it seems to me this is is a hack so you can keep the `map` at 1359. In this case it might be simpler to just replace that map with a `for ... of` or `forEach` and push the declaration plus an additional optional push for the value to some predefined list.

Reply:
I won't deny it was a bit of a hack. I've refactored this into something a bit more sensible.